magpie/magpieadapter/__init__.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import os
+import sys
+
+# TODO Is the following block really useful?
+this_dir = os.path.abspath(os.path.dirname(__file__))
+sys.path.insert(0, this_dir)
+
+
+from twitcher.adapter.base import AdapterInterface
+from twitcher.owsproxy import owsproxy
+
+
+logger = logging.getLogger(__name__)
+
+
+__version__ = '0.1.1'
+
+
+class MagpieAdapter(AdapterInterface):
+
+    def servicestore_factory(self, registry, database=None, headers=None):
+        from magpie.magpieadapter.magpieservice import MagpieServiceStore
+        return MagpieServiceStore(registry=registry, headers=headers)
+
+    def owssecurity_factory(self, registry):
+        from magpie.magpieadapter.magpieowssecurity import MagpieOWSSecurity
+        # TODO For magpie we cannot store the servicestore object since the constructor need a header with token
+        # taken from the request... maybe we should check for that?!?
+        #return MagpieOWSSecurity(tokenstore_factory(registry), servicestore_factory(registry))
+        return MagpieOWSSecurity()
+
+    def configurator_factory(self, settings):
+        from pyramid.config import Configurator
+        from pyramid.authentication import AuthTktAuthenticationPolicy
+        from pyramid.authorization import ACLAuthorizationPolicy
+        from ziggurat_foundations.models import groupfinder
+
+        magpie_secret = settings['magpie.secret']
+
+        # Disable rpcinterface which is conflicting with postgres db
+        settings['twitcher.rpcinterface'] = False
+
+        authn_policy = AuthTktAuthenticationPolicy(
+            magpie_secret,
+            callback=groupfinder,
+        )
+        authz_policy = ACLAuthorizationPolicy()
+
+        config = Configurator(
+            settings=settings,
+            authentication_policy=authn_policy,
+            authorization_policy=authz_policy
+        )
```

Comment:
from `authn_policy` (L44) to here is identical to [magpie main L131-L142](https://github.com/Ouranosinc/Magpie/blob/5fde2f85743ce7cdbb2b0821d9be4cae777a99fd/magpie/magpie.py#L131-L142)
could be a nice little function defined in `security.py`